cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -92,6 +93,13 @@ func Run(gadgetRoot, device string, options Options) error {
 	if err := diskLayout.RemoveCreated(); err != nil {
 		return fmt.Errorf("cannot remove partitions from previous install: %v", err)
 	}
+	// at this point we removed any existing partition, nuke any of the
+	// existing keyfiles too (LP: #1879338)
+	for _, p := range []string{options.KeyFile, options.RecoveryKeyFile} {
```

Comment:
the recovery key file is usually inside the partitions, so not sure it makes sense to remove it that way.

Also we should probably ignore them directly if they are empty

Reply:
Thanks, excellent points. Updated and force-pushed for easier cherry-picking.